api/src/main/java/org/openmrs/util/OpenmrsUtil.java

```diff
@@ -280,37 +280,19 @@ public static void copyFile(InputStream inputStream, OutputStream outputStream)
 				catch (Exception e) { /* pass */
 				}
 			}
-			
+
 			return;
 		}
-		
-		InputStream in = null;
-		OutputStream out = null;
-		try {
-			in = new BufferedInputStream(inputStream);
-			out = new BufferedOutputStream(outputStream);
-			while (true) {
-				int data = in.read();
-				if (data == -1) {
-					break;
-				}
-				out.write(data);
-			}
-		}
-		finally {
-			if (in != null) {
-				in.close();
-			}
-			if (out != null) {
-				out.close();
-			}
-			try {
-				outputStream.close();
-			}
-			catch (Exception e) { /* pass */
-			}
+
+		try (
+
+				AutoCloseable closeable = () -> IOUtils.copy(inputStream, outputStream)
```

Comment:
the javadoc says the "outputStream /is/ closed at completion of this method" the inputstream is not.

I dont think your implementation using the try-with-resource block closes the outputStream.

you could try:

```java
 try {
     IOUtils.copy(inputStream, outputStream);
 } finally {
     IOUtils.closeQuietly(outputStream);
 }
```

see https://commons.apache.org/proper/commons-io/javadocs/api-release/org/apache/commons/io/IOUtils.html

also please write test cases for all possible cases. for example the logic at the top where it closes the outputStream if the inputStream is null is not tested.